src/mds/PurgeQueue.cc

```diff
@@ -141,6 +142,22 @@ void PurgeQueue::activate()
 {
   std::lock_guard l(lock);
 
+  {
+    PurgeItem item;
+    bufferlist bl;
+
+    ::encode(item, bl);
+    purge_item_journal_size = bl.length() + journaler.get_journal_envelope_size(); 
+
+    uint64_t write_pos = journaler.get_write_pos(); 
+    uint64_t expire_pos = journaler.get_expire_pos(); 
+    uint64_t item_num = (write_pos - expire_pos) / purge_item_journal_size;
+    dout(4) << "PurgeQueue item_size=" << purge_item_journal_size 
+      << " write_pos=" << write_pos << " expire_pos=" << expire_pos 
+      << " num_queued=" << item_num << dendl;
+    logger->set(l_pq_item_in_journal, item_num);
```

Comment:
Do we need to repeat this code? Why not just only seet it in `_execute_item_complete`? Keep the debug message too but I think `dout(10)` or 7 is more appropriate.

Reply:
sounds right :)